### PR TITLE
Add EkinoNewrelicBundle

### PR DESCRIPTION
**Pull request by GromNaN**
* Is it possible to add conditional configuration when an other bundle is installed? This bundle has interactions with [SonataBlockBundle](https://github.com/ekino/EkinoNewRelicBundle#integration-with-sonatablockbundle).
* Configuration could be made interactive with questions like "What transaction naming do you want to use?" or "Are you using Symfony HTTP Cache?"

### CODE REVIEW

File: ekino/newrelic-bundle/1.3/manifest.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "bundles": {
+        "Ekino\\Bundle\\NewRelicBundle\\NewRelicBundle": ["all"]
```
Comment (fabpot) [inline ekino/newrelic-bundle/1.3/manifest.json]: Do we really want to enable it for all envs? Or just the `prod` one?

**Comment by fabpot** [comment]
There is no difference between configuring a bundle with a recipe or by hand. So, the answer is no to the "conditional configuration". That should be done via a compiler pass.

I'm against interactive questions as the goal is to **auto**configure things. You can display some information about the next steps at the end of the install if needed. But that should not replace proper documentation.

File: ekino/newrelic-bundle/1.3/manifest.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    },
+    "copy-from-recipe": {
+        "etc/": "%ETC_DIR%/"
+	},
```
Comment (fabpot) [inline ekino/newrelic-bundle/1.3/manifest.json]: Not valid JSON

File: ekino/newrelic-bundle/1.3/manifest.json (hunk 1 above)
Comment (GromNaN) [inline ekino/newrelic-bundle/1.3/manifest.json]: Indeed, I'm changing it and moving the config file to prod directory.

File: ekino/newrelic-bundle/1.3/manifest.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "bundles": {
+        "Ekino\\Bundle\\NewRelicBundle\\NewRelicBundle": ["prod"]
+    },
+    "copy-from-recipe": {
+        "etc/": "%ETC_DIR%/"
+	}
```
Comment (fabpot) [inline ekino/newrelic-bundle/1.3/manifest.json]: Looks like there is an indent issue

File: ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+ekino_new_relic:
```
Comment (GromNaN) [inline ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml]: Naming the yaml file with the config root key could be considered as a best practice.

**Comment by fabpot** [comment]
This one is quite interesting as this is one that could be in the "main" repository. /cc @symfony/deciders

File: ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml (hunk 4 above)
Comment (fabpot) [inline ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml]: Indeed, I will add a check for that in the validator.

File: ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    # enabled: true                         # Defaults to true
+    # application_name: PHP Application     # default value in newrelic is "PHP Application", or whatever is set as php ini-value
+    # deployment_names: ~                   # default value is 'application_name', supports string array or semi-colon separated string
+    # api_key: "%env(NEWRELIC_API_KEY)%"    # New Relic API
+    # license_key:                          # New Relic license key (optional, default value is read from php.ini)
+    # xmit: false                           # if you want to record the metric data up to the point newrelic_set_appname is called, set this to true (default: false)
+    # logging: false                        # If true, logs all New Relic interactions to the Symfony log (default: false)
+    # instrument: false                     # If true, uses enhanced New Relic RUM instrumentation (see below) (default: false)
```
Comment (javiereguiluz) [inline ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml]: The *"(see below)"* mention in this line, what does it refer to?

**Comment by javiereguiluz** [comment]
In my opinion, before considering to include this bundle in the main repo, the documentation should be completely updated. It only talks about "Symfony2" and it evens explains how to install the bundle using the "deps" file of Symfony 2.0.x.

File: ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml (hunk 5 above)
Comment (GromNaN) [inline ekino/newrelic-bundle/1.3/etc/packages/prod/ekino_new_relic.yaml]: Copy/paste from the readme.

**Comment by stof** [comment]
I don't think the file in the recipe should have all available options commented. It should suggest only the important ones (`api_key` for instance).
Advanced options are useless for most people and so should not be here. Th doc is the place where they should be explained

**Comment by iltar** [comment]
What is exactly the workflow to get something into the main repository? symfony/recipes#29

**Comment by Pierstoval** [comment]
> What is exactly the workflow to get something into the main repository? symfony/recipes#29

It will be decided by the core team (a.k.a @symfony/deciders) , I think.

**Comment by GromNaN** [comment]
Good idea @stof. I've updated the PR.
I agree that this bundle need some cleanup.

**Comment by fabpot** [comment]
👍

**Comment by rande** [comment]
I have updated the readme file to remove the obsolete deps installation reference.

Thanks

**Pull request closed without merging**